### ISSUE TITLE
Versioning scheme

### ISSUE THREAD

**Issue by ptomato** [gjsify/ts-for-gir#187]
What versioning scheme should this module follow? An alternating stable-unstable one like GNOME does, so that you can derive the corresponding GJS version directly from the module's version? or semver, where you bump the major version every time a new version of the module can't be used with previous GJS, and there's like a table in the docs that says "4.x ↔ GJS 1.76–82" or whatever.

**Comment by JumpLink** [comment]
Good point. It should definitely be documented and maintained somewhere. Maybe we can tag types in the comments with @version @deprecated tags and then support a wide range of GJS versions and only remove old types for major breaking changes in GJS? I'm still completely open here
